README.md

```diff
@@ -187,6 +188,37 @@ This little example should show how you can use the middlwares e.g. to check or
 
 Checkout the `examples/middleware` how to add multiple middlewares.
 
+### Streaming body data
+
+If you have operations that need time to be computed(of course non-blocking ;-)), you can stream the intermediary results directly to the client,
+without buffering on your side.
+
+Use an instance of the `HttpBodyStream` and use this instance as the body for `Response` object you want to return.
+
+```php
+$callback = function (RequestInterface $request) {
+    $input = new ReadableStream();
+    $body = new HttpBodyStream($input);
+    
+    // your computation
+    // emit via `$input`
+    
+    return new Response(
+        200,
+        array(),
+        $body
+    );
+}
+```
+
+The `HttpServer` will use the emitted data from the `ReadableStream` to send this data directly to the client.
+
+You can add an the optional encoder as second parameter of the constructor of the `HttpBodyStream`. If none is set the default will the be`ChunkedEncoderStream` an 
+encoder which will send HTTP encoded chunks to the client.
+The `HttpServer` will automaticly add a 'Transfer-Encoding: chunked' line to your header, if you use the `HttpBodyStream` with the default `ChunkedEncoderStream`.
```

Comment:
This is a fairly good documentation about the *what*, what's more important to consumers of this library is probably the *why*